### PR TITLE
Nice up the function (section)

### PR DESCRIPTION
**Pull request by Robsteranium**
- removed references to turtle functions
- reintroduced total-bill etc instead
- explain predicate functions (not just about names)
- explain anonymous functions as useful predicates
- explain side-effects (to those that get this far)

### CODE REVIEW

File: outline/functions.md [hunk 1]
```diff
@@ -12,171 +12,83 @@ http://clojurebridge.github.io/curriculum/outline/functions.html
 
 {% endcomment %}
 
-<section>
+<section ng-controller="NarrativeController">
```
Comment (plexus) [inline outline/functions.md:15]: What does this do?

Comment (Robsteranium) [inline outline/functions.md:15]: I actually don't know. I just saw it all over the newer slides so I replicated it.

Comment (Robsteranium) [inline outline/functions.md:15]: It looks like it changes how the details links work: https://github.com/clojurebridge-berlin/curriculum/blob/gh-pages/assets/script.js

**Comment by plexus** [comment]
I had a walk through the slides. It's certainly an improvement. I do feel there are a lot of gaps that the coaches will have to fill in. If you're coaching absolute beginners then this section will be quite overwhelming I think. Concepts are introduced one by one, but they're not really broken up, and there's not much repetition and recap.

**Pull request merged**